clusterloader2/pkg/measurement/common/network-policy/network-policy-enforcement-latency.go

```diff
@@ -0,0 +1,568 @@
+	case "setup":
+		return nil, nps.setup(config)
+	case "run":
+		return nil, nps.run(config)
+	case "complete":
+		return nil, nps.complete(config)
+	default:
+		return nil, fmt.Errorf("unknown action %v", action)
+	}
+}
+
+// setup initializes the measurement, creates a namespace, network policy egress
+// allow to the kube-apiserver and permissions for the network policy
+// enforcement test clients.
+func (nps *networkPolicyEnforcementMeasurement) setup(config *measurement.Config) error {
+	err := nps.initializeMeasurement(config)
+	if err != nil {
+		return fmt.Errorf("failed to initialize the measurement: %v", err)
+	}
+
+	if err := client.CreateNamespace(nps.k8sClient, nps.testClientNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+
+	// Create network policies for non-baseline test.
+	if !nps.baseline {
+		if err = nps.createPolicyAllowAPIServer(); err != nil {
+			return err
+		}
+
+		// Create a policy that allows egress from pod creation test client pods to
+		// target pods.
+		podCreationAllowPolicyName := fmt.Sprintf("%s-pod-creation", allowPolicyName)
+		if err = nps.createPolicyToTargetPods(podCreationAllowPolicyName, "", podCreationTest, true); err != nil {
+			return err
+		}
+
+		// Create a policy that denies egress from policy creation test client pods
+		// to target pods.
+		policyCreationDenyPolicyName := fmt.Sprintf("%s-policy-creation", denyPolicyName)
+		if err = nps.createPolicyToTargetPods(policyCreationDenyPolicyName, "", policyCreationTest, false); err != nil {
+			return err
+		}
+	}
+
+	return nps.createPermissionResources()
+}
+
+func (nps *networkPolicyEnforcementMeasurement) initializeMeasurement(config *measurement.Config) error {
+	if nps.framework != nil {
+		return fmt.Errorf("the %q is already started. Cannot start again", networkPolicyEnforcementName)
+	}
+
+	var err error
+	if nps.targetLabelKey, err = util.GetString(config.Params, "targetLabelKey"); err != nil {
+		return err
+	}
+
+	if nps.targetLabelValue, err = util.GetString(config.Params, "targetLabelValue"); err != nil {
+		return err
+	}
+
+	if nps.testClientNamespace, err = util.GetStringOrDefault(config.Params, "testClientNamespace", netPolicyTestNamespace); err != nil {
+		return err
+	}
+
+	if nps.baseline, err = util.GetBoolOrDefault(config.Params, "baseline", false); err != nil {
+		return err
+	}
+
+	if nps.testClientNodeSelectorKey, err = util.GetString(config.Params, "testClientNodeSelectorKey"); err != nil {
+		return err
+	}
+
+	if nps.testClientNodeSelectorValue, err = util.GetString(config.Params, "testClientNodeSelectorValue"); err != nil {
+		return err
+	}
+
+	nps.framework = config.ClusterFramework
+	nps.k8sClient = config.ClusterFramework.GetClientSets().GetClient()
+
+	namespaceList, err := nps.k8sClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Target namespaces are only those that have a "test-" prefix.
```

Comment:
Shouldn't this be `framework.GetAutomanagedNamespacePrefix()`?

Reply:
It should. Fixed.